script.metadata.editor/resources/lib/rating_updater.py

```diff
@@ -464,7 +501,101 @@ def _update_uniqueid_dict(self,key,value):
         self.update_uniqueid = True
 
     def _set_value(self,key,value):
-        json_call('VideoLibrary.Set%sDetails' % self.dbtype,
-                  params={key: value, '%sid' % self.dbtype: int(self.dbid)},
-                  debug=LOG_JSON
-                  )
\ No newline at end of file
+        self.db.write(key=key, value=value)
+
+    def _omdb(self):
+        if not OMDB_API:
+            log('No OMDb API key configured. Skip.', force=RATING_DEBUG)
+            return
+
+        if self.imdb:
+            url = 'http://www.omdbapi.com/?apikey=%s&i=%s&plot=short&r=xml&tomatoes=true' % (OMDB_API, self.imdb)
+
+        elif OMDB_FALLBACK and self.dbtype != 'episode' and self.original_title and self.year:
+            # urllib has issues with some asian letters
+            try:
+                title = urllib.quote(self.original_title)
+            except KeyError:
+                return
+
+            url = 'http://www.omdbapi.com/?apikey=%s&t=%s&year=%s&plot=short&r=xml&tomatoes=true' % (OMDB_API, title, self.year)
+
+        else:
+            return
+
+        error_msg = 'OMDb error for "%s (%s)" IMDBd "%s". Error --> ' % (self.original_title, self.year, self.imdb)
+
+        for i in range(1,4): # loop if heavy server load
+            log('OMDb call try %s/3' % str(i), force=RATING_DEBUG)
+            try:
+                request = requests.get(url, timeout=5)
+                if not str(request.status_code).startswith('5'):
+                    break
+                elif i == 1:
+                    notification('OMDb', ADDON.getLocalizedString(32024))
+
+            except Exception:
+                if i < 3:
+                    xbmc.sleep(500)
+                else:
+                    log(error_msg + '408', WARNING)
+                    return
+
+        if request.status_code == 401:
+            log('OMDb error --> API limit reached', WARNING)
+            if DIALOG.yesno(xbmc.getLocalizedString(257), ADDON.getLocalizedString(32033)):
+                log('OMDb limit reached and disabled for next calls', force=RATING_DEBUG)
+                self.omdb_limit = True
+            else:
+                log('OMDb limit reached and rating updater canceled', force=RATING_DEBUG)
+                winprop('CancelRatingUpdater.bool', True)
+            return
+
+        elif request.status_code != requests.codes.ok:
+            log(error_msg + str(request.status_code), WARNING)
+            return
+
+        result = request.content
```

Comment:
`.content` property returns a byte string, not text. Are you sure that you need bytes instead of unicode text?

Reply:
works both. Is there any benefit from switching to `.text` ?